### PR TITLE
[FIX] #467 django 1.10 manage.py startup error

### PR DESCRIPTION
**Pull request by yjmade**
see https://github.com/celery/django-celery/issues/462#issuecomment-245531631

### CODE REVIEW

**Comment by auvipy** [comment]
thanks!! could you please add test and changlog too?

**Comment by auvipy** [comment]
build is failing

**Comment by azmeuk** [comment]
A good idea would be to use `python -Wall` instead of just `python` to launch tests in `tox.ini`. This way, tests will always show django deprecations. And it will be easier to fix them before django releases are out, and the library is outdated.

**Comment by yjmade** [comment]
now it's passed on all django1.10 and failed on all others. 
@azmeuk  I tried ```python -Wall```, still fail for none-1.10

**Comment by azmeuk** [comment]
@yjmade this is normal, `-Wall` just displays warnings and deprecations.

**Comment by auvipy** [comment]
builds are still failing

File: djcelery/tests/test_commands.py [hunk 1]
```diff
@@ -14,11 +14,11 @@ def test_celeryd_command():
     else:
         traceback = None
     with patch(CELERYD_COMMAND) as handle:
-        call_command('celeryd')
+        call_command('celeryd', '--hostname=test', '--loglevel=WARN')
```
Comment (vytisb) [inline djcelery/tests/test_commands.py]: `execute_from_command_line(['manage.py', 'celeryd', '--hostname=test', '--loglevel=WARN'])`
And remove `skip_checks=True` below.

**Comment by azmeuk** [comment]
I think there is another error here.
Django 1.10 [drops the support of optparse](https://docs.djangoproject.com/en/1.9/howto/custom-management-commands/) in favor of argparse. But celeryd command use `celery.bin.worker` function, that returns a optparse `Option` object in versions 3.1 and 4.0.

There was a [bug opened](https://github.com/celery/celery/issues/793) in 2012 but it has been closed and nothing happened.

Two solutions here:
- Convince upstream, and make pull requests for celery 3.1 and 4.0 so they use argparse.
- Find a workaround so celeryd can use argparse.

I might have a look at the second solution.

**Comment by vytisb** [comment]
@azmeuk, we do have a workaround. Have a look [here](https://github.com/celery/django-celery/blob/79d9689b62db3d54ebd0346e00287f91785f6355/djcelery/management/base.py#L61-L80).

But it's not perfect and this PR aims to improve it.

**Pull request closed without merging**